src/widget/sidebar_filler.c

```diff
@@ -0,0 +1,295 @@
+#include "sidebar_filler.h"
+
+#include "building/menu.h"
+#include "city/labor.h"
+#include "city/message.h"
+#include "city/population.h"
+#include "city/ratings.h"
+#include "city/view.h"
+#include "city/warning.h"
+#include "core/config.h"
+#include "core/direction.h"
+#include "core/lang.h"
+#include "core/string.h"
+#include "game/orientation.h"
+#include "game/settings.h"
+#include "game/state.h"
+#include "game/undo.h"
+#include "graphics/arrow_button.h"
+#include "graphics/graphics.h"
+#include "graphics/image.h"
+#include "graphics/image_button.h"
+#include "graphics/lang_text.h"
+#include "graphics/menu.h"
+#include "graphics/panel.h"
+#include "graphics/screen.h"
+#include "graphics/text.h"
+#include "graphics/window.h"
+#include "map/orientation.h"
+#include "scenario/property.h"
+#include "scenario/criteria.h"
+#include "sound/effect.h"
+#include "widget/city.h"
+#include "widget/minimap.h"
+#include "window/advisors.h"
+#include "window/build_menu.h"
+#include "window/city.h"
+#include "window/empire.h"
+#include "window/message_dialog.h"
+#include "window/message_list.h"
+#include "window/mission_briefing.h"
+#include "window/overlay_menu.h"
+
+#define EXTRA_INFO_LINE_SPACE 32
+#define EXTRA_INFO_HEIGHT_GAME_SPEED 64
+#define EXTRA_INFO_HEIGHT_UNEMPLOYMENT 112
+#define EXTRA_INFO_HEIGHT_RATINGS 272
+#define EXTRA_INFO_TOP_PADDING 10;
+#define EXTRA_INFO_SPEED_PADDING 26;
+#define EXTRA_INFO_UNEMPLOYMENT_TOP_PADDING 20;
+#define EXTRA_INFO_UNEMPLOYMENT_BOTTOM_PADDING 22;
+
+static void button_game_speed(int is_down, int param2);
+
+static arrow_button arrow_buttons_speed[] = {
+    {11, 30, 17, 24, button_game_speed, 1, 0},
+    {35, 30, 15, 24, button_game_speed, 0, 0},
+};
+
+typedef struct {
+    int value;
+    int target;
+} objective;
+
+static struct {
+    int height;
+    int game_speed;
+    int unemployment_percentage;
+    int unemployment_amount;
+    objective culture;
+    objective prosperity;
+    objective peace;
+    objective favor;
+    objective population;
+} data;
+
+int sidebar_filler_is_enabled(void)
+{
+    int enabled = config_get(CONFIG_UI_SIDEBAR_INFO);
+    return enabled;
+}
+
+int sidebar_filler_get_filler_height(void)
+{
+    return data.height;
+}
+
+int sidebar_filler_handle_mouse(const mouse *m, int x_offset, int y_offset)
+{
+    int click = 0;
+    click |= arrow_buttons_handle_mouse(m, x_offset, y_offset, arrow_buttons_speed, 2);
+    return click;
+}
+
+static int sidebar_filler_calculate_extra_info_height(int y_offset, int is_collapsed)
+{
+    if (is_collapsed || !config_get(CONFIG_UI_SIDEBAR_INFO)) {
+        data.height = 0;
+    } else {
+        int available_height = screen_height() - y_offset;
```

Comment:
Using `screen_height()` implicitly assumes that the sidebar flows to the edge of the screen. That might not be true in the future (and wasn't true in the past). Better pass the available height instead.